src/pages/lessons/[slug].tsx

```diff
@@ -182,7 +113,22 @@ const lessonLoader = (slug: any, token: any) => (query: string) => {
       ...authorizationHeader,
     },
   })
-  return graphQLClient.request(query, variables)
+  return graphQLClient.request(lessonQuery, variables)
+}
+
+const NextResourceButton: FunctionComponent<{
```

Comment:
I made this component as a small example of composition to consider. If the component has children, they are delivered and displayed, if not is uses default text `Next Lesson`